jobserver/views/projects.py

```diff
@@ -211,7 +211,13 @@ def form_valid(self, form):
     def get_object(self):
         project = get_object_or_404(Project, slug=self.kwargs["project_slug"])
 
-        if not project.members.filter(username=self.request.user.username).exists():
+        user = self.request.user
+        user_is_member = project.members.filter(username=user.username).exists()
+        user_has_project_manage = has_permission(
+            user, permissions.project_manage, project=project
+        )
+
+        if not user_is_member or not user_has_project_manage:
```

Comment:
At the moment, this raises if someone has the global ProjectDeveloper role but isn't a member of the project. However, the intent of the global role is that you don't need to be a member of the project.

I think you can remove the `not user_is_member` half of this if-statement and it will work as expected. Although you should probably add another test; something like `test_projectedit_get_success_with_global_role`.

Reply:
Originally, only members of the project could edit this view. The issue calls for restricting editing this view to a user with the Project Developer role. Arguably, that's less restrictive, although easier to reason about.

Reply:
Thanks for talking this through, @lucyb. I've updated the `get_object` method, which now restricts only using the `project_manage` permission. I've added an extra test for users that have global `project_manage`.